### PR TITLE
Dynamically defining exports from common

### PR DESCRIPTION
**Pull request by aarongustafson**
Related to #84, I had an idea that would not require a wholesale overhaul of the definition export. Curious for your thoughts @marcoscaceres @jnurthen

This commit:

* Defines the terms to be exported in a variable `definitions_to_export`
* Parses the "common" include using the `oninclude` hook and adds the `data-export` param to the existing `dfn` elements that match the values in that array.

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Aug 25, 2020, 8:06 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [Spec Generator](https://www.w3.org/2015/labs/) - Spec Generator is the web service used to build specs that rely on ReSpec.

:link: [Related URL](https://labs.w3.org/spec-generator/?type=respec&url=https%3A%2F%2Frawcdn.githack.com%2Fw3c%2Faccname%2F80f82400d396e8239b8632fdefc91dc433c3d069%2Findex.html%3FisPreview%3Dtrue)

```

😭  Sorry, there was an error generating the HTML. Please report this issue!
[36mSpecification: https://rawcdn.githack.com/w3c/accname/80f82400d396e8239b8632fdefc91dc433c3d069/index.html?isPreview=true&publishDate=2020-08-25[39m
[36mReSpec version: 25.6.0[39m
[36mFile a bug: https://github.com/w3c/respec/[39m
[36mError: Error: Evaluation failed: Timeout: document.respecIsReady didn't resolve in 29054ms.[39m
[36m    at ExecutionContext._evaluateInternal (/u/spec-generator/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:217:19)[39m
[36m    at runMicrotasks (<anonymous>)[39m
[36m    at processTicksAndRejections (internal/process/task_queues.js:97:5)[39m
[36m    at async ExecutionContext.evaluate (/u/spec-generator/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:106:16)[39m
[36m    at async generateHTML (/u/spec-generator/node_modules/respec/tools/respecDocWriter.js:128:12)[39m
[36m    at async fetchAndWrite (/u/spec-generator/node_modules/respec/tools/respecDocWriter.js:95:18)[39m
[36m    at async Object.generate [as respec] (/u/spec-generator/generators/respec.js:14:20)[39m
[36m    at async generate (/u/spec-generator/server.js:90:29)[39m
[36m[39m
```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20w3c/accname%2385.)._
</details>

### CODE REVIEW

**Comment by aarongustafson** [comment]
If this approach is workable for you @jnurthen, it could be employed on any/all ARIA specs.

**Comment by jnurthen** [comment]
we can't move to use data-export until  https://github.com/w3c/respec/issues/2825 is resolved

**Comment by marcoscaceres** [comment]
> we can't move to use data-export until w3c/respec#2825 is resolved

Just noting that the dfn panel is a consumer of the definitions, but it shouldn't block this.

**Comment by jnurthen** [comment]
@marcoscaceres I disagree. I cannot ship a spec with known accessibility errors in it.

**Comment by marcoscaceres** [comment]
Ok, I'll look at adding a feature so you can disable the dfn panel in this spec. We can re-enable it once the aforementioned bug is fixed. Note, however, you are disadvantaging a larger set of users refusing to ship this even if the panel is a bit buggy - the panel still extremely useful for a significant number of users even if it can't currently be activated by keyboard.

**Comment by marcoscaceres** [comment]
@jnurthen and I spoke on Slack about  w3c/respec#2825 - Going to get some guidance on how best to fix  w3c/respec#2825  from @jnurthen over on the other repo. I put up some suggestions. 

@aarongustafson, I'll try to review the code + changes here too ASAP.

File: index.html [hunk 1]
```diff
@@ -140,12 +137,31 @@
...
+    while( i-- )
+    {
+      var text = $dfns[i].innerText.toLowerCase();
+      if ( definitions_to_export.indexOf( text ) > -1 )
+      {
+        $dfns[i].dataset.export = "";
+      }
+    }
+    return $dfnDiv.innerHTML;
+  }
```
Comment (marcoscaceres) [inline index.html:161]:
```suggestion
   function exportDefinitions(_, content) {
    const dfnsToExport = ["accessible name", "accessible description"];
    const temp = document.createElement("temp");
    temp.innerHTML = content;

    const dfns = [
      ...temp.querySelectorAll("dfn:not([data-export])"),
    ].filter(({ textContent: text }) =>
      dfnsToExport.includes(text.toLowerCase().trim())
    );

    dfns.forEach(dfn => (dfn.dataset.export = ""));
    return temp.innerHTML;
  }
```

**Comment by marcoscaceres** [comment]
I think this might be ok for a transitional step.

File: index.html [hunk 2]
```diff
@@ -3,11 +3,8 @@
 <head>
 <title>Accessible Name and Description Computation 1.2</title>
 <meta content="text/html; charset=utf-8" http-equiv="Content-Type"/>
-<script src="https://www.w3.org/Tools/respec/respec-w3c-common" class="remove" defer></script>
```
Comment (marcoscaceres) [inline index.html:6]: Probably shouldn't touch any of these.

File: index.html [hunk 3]
```diff
@@ -140,12 +137,31 @@
        "FPWD": "https://www.w3.org/TR/core-aam-1.2/",
        "REC": "https://www.w3.org/TR/core-aam-1.2/"
     },
-
-    localBiblio: biblio,
-    
     preProcess: [ linkCrossReferences ]
   }
 </script>
+<script>
```
Comment (marcoscaceres) [inline index.html:143]:
```suggestion
<script class="remove">
```

**Comment by jnurthen** [comment]
Defintions have now been moved to their source specs. Most are defined in  "wai-aria" and a few in "core-aam"

I think this can be closed.

**Comment by marcoscaceres** [comment]
Thanks for the update @jnurthen!

**Pull request closed without merging**